lib/portage/package/ebuild/doebuild.py

```diff
@@ -2131,9 +2131,35 @@ def spawn(
         logname_backup = mysettings.configdict["env"].get("LOGNAME")
         mysettings.configdict["env"]["LOGNAME"] = logname
 
+    eapi = mysettings["EAPI"]
+
+    # TODO: Move into eapi.py
+    def eapi_does_not_export_a(eapi):
+        return eapi == "9"
+
+    dont_export_a = "dont-export-a" in mysettings.features or eapi_does_not_export_a(
+        eapi
+    )
+
+    if dont_export_a:
+        orig_env = mysettings.environ()
+        # Copy since we are potentially removing keys from the dict.
+        env = orig_env.copy()
+        ebuildExtraSource = os.path.join(env["T"], "portage-ebuild-extra-source")
+        with open(ebuildExtraSource, mode="w") as f:
+            for name, value in orig_env.items():
+                if name != "A":
```

Comment:
I realize that this is not optimal, but that's a leftover from a previous implementation where everything was passed via file and not via `env`. This should be changed once this PR leaves the PoC stage.